### PR TITLE
Remove Releng from go.mod, go.sum, and .changeset CODEOWNERS paths

### PR DESCRIPTION
**Pull request by chainchad**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # Root
 * @smartcontractkit/foundations
 
-.changeset @smartcontractkit/releng @smartcontractkit/foundations
+.changeset @smartcontractkit/foundations
```
Comment (chainchad) [inline .github/CODEOWNERS]: @jmank88 would you be good w/ this?

Comment (patrickhuie19) [reply .github/CODEOWNERS]: FWIW, I'd be happy with https://github.com/orgs/smartcontractkit/teams/engineers here and for go.mod + go.sum

**Review by erikburt — DISMISSED** [review]

**Review by momentmaker — DISMISSED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (jmank88) [reply .github/CODEOWNERS]: We just added releng here because we needed more reviewers. How about core?

Comment (chainchad) [reply .github/CODEOWNERS]: Added core in 40a39d79777a10c455235fdfb8a6724bde413e16

**Review by momentmaker — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**